### ISSUE TITLE
Villagers not spawning naturally and existing villagers disappearing

### ISSUE THREAD

**Issue by alliumm** [Estecka/mc-Shifting-Wares#4]
To reproduce: create a new world, run `/locate structure village_plains`, and teleport to the village. No villagers spawn naturally. The bug is not present in v1.0.1.

I didn't notice any abnormal log messages when generating villages, but loading a world with existing villagers logs this message every time one attempts to load.
```
[14:30:00] [Server thread/WARN]: Exception loading entity: 
net.minecraft.class_148: Loading entity NBT
	at net.minecraft.class_1297.method_5651(class_1297.java:1939) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_17839(class_1299.java:549) ~[client-intermediary.jar:?]
	at net.minecraft.class_156.method_17974(class_156.java:513) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_5892(class_1299.java:548) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_17848(class_1299.java:635) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299.method_17842(class_1299.java:586) ~[client-intermediary.jar:?]
	at net.minecraft.class_1299$1.method_31490(class_1299.java:609) ~[client-intermediary.jar:?]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1856) ~[?:?]
	at net.minecraft.class_1299$1.tryAdvance(class_1299.java:608) ~[client-intermediary.jar:?]
	at java.util.Spliterator.forEachRemaining(Spliterator.java:332) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
	at net.minecraft.class_5565.method_31731(class_5565.java:76) ~[client-intermediary.jar:?]
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_3215$class_4212.method_18859(class_3215.java:564) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_3215$class_4212.method_16075(class_3215.java:573) ~[client-intermediary.jar:?]
	at net.minecraft.class_3215.method_19492(class_3215.java:282) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_20415(MinecraftServer.java:5824) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16075(MinecraftServer.java:758) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_16208(MinecraftServer.java:742) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:675) ~[client-intermediary.jar:?]
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:265) ~[client-intermediary.jar:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.lang.AbstractMethodError: Receiver class net.minecraft.class_1646 does not define or inherit an implementation of the resolved method 'abstract net.minecraft.class_1916 getOffers()' of interface tk.estecka.shiftingwares.IVillagerEntityDuck.
	at tk.estecka.shiftingwares.MapTradesCache.FillCacheFromTrades(MapTradesCache.java:17) ~[shifting-wares-1.0.2+1.20.1.jar:?]
	at net.minecraft.class_1646.handler$cam000$shifting-wares$ReadCachedMapsFromNbt(class_1646.java:1705) ~[client-intermediary.jar:?]
	at net.minecraft.class_1646.method_5749(class_1646.java:569) ~[client-intermediary.jar:?]
	at net.minecraft.class_1297.method_5651(class_1297.java:1930) ~[client-intermediary.jar:?]
	... 29 more
```

**Comment by Estecka** [comment]
I think I've identified the issue. Unfortunately that one only happens in prod, not in dev.

I've taken down 1.0.2 and 1.0.3 for now.

**Comment by Estecka** [comment]
Fixed in 1.0.4, up on modrinth.
